# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/ExternalHelperAppParent.h
# Commit: baffa8a7b961
# Full Hash: baffa8a7b9618e9a02e4bd808adb71dead99cce5
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-11-12 09:43:07
# Regressor Bug: 1589270
# File Overlap Count: 1
# Description:
#   Bug 1589270 - Part 3: Convert nsExternalHelperApp to use BrowsingContext instead of nsIInterfaceRequestor. r=bzbarsky
#   
#   This also converts MaybeCloseWindowHelper, and results in the window close operations being always run in the parent (even without DocumentChannel).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49528
# ==============================================================================

diff -r 68bc2ae9b89a -r baffa8a7b961 uriloader/exthandler/ExternalHelperAppParent.h
--- a/uriloader/exthandler/ExternalHelperAppParent.h	Fri Nov 08 03:01:03 2019 +0000
+++ b/uriloader/exthandler/ExternalHelperAppParent.h	Fri Nov 08 04:35:05 2019 +0000
@@ -17,6 +17,8 @@
 class URI;
 }  // namespace IPC
 
+class nsExternalAppHandler;
+
 namespace mozilla {
 
 namespace ipc {
@@ -70,15 +72,15 @@
   NS_DECL_NSISTREAMLISTENER
   NS_DECL_NSIREQUESTOBSERVER
 
-  mozilla::ipc::IPCResult RecvOnStartRequest(const nsCString& entityID,
-                                             PBrowserParent* aBrowser) override;
+  mozilla::ipc::IPCResult RecvOnStartRequest(
+      const nsCString& entityID) override;
   mozilla::ipc::IPCResult RecvOnDataAvailable(const nsCString& data,
                                               const uint64_t& offset,
                                               const uint32_t& count) override;
   mozilla::ipc::IPCResult RecvOnStopRequest(const nsresult& code) override;
 
   mozilla::ipc::IPCResult RecvDivertToParentUsing(
-      PChannelDiverterParent* diverter, PBrowserParent* aBrowser) override;
+      PChannelDiverterParent* diverter) override;
 
   bool WasFileChannel() override { return mWasFileChannel; }
 
@@ -91,7 +93,7 @@
   void Init(const Maybe<mozilla::net::LoadInfoArgs>& aLoadInfoArgs,
             const nsCString& aMimeContentType, const bool& aForceSave,
             const Maybe<mozilla::ipc::URIParams>& aReferrer,
-            PBrowserParent* aBrowser);
+            BrowsingContext* aContext, const bool& aShouldCloseWindow);
 
  protected:
   virtual ~ExternalHelperAppParent();
@@ -100,7 +102,7 @@
   void Delete();
 
  private:
-  nsCOMPtr<nsIStreamListener> mListener;
+  RefPtr<nsExternalAppHandler> mListener;
   nsCOMPtr<nsIURI> mURI;
   nsCOMPtr<nsILoadInfo> mLoadInfo;
   bool mPending;